### PR TITLE
Migrate documentation to sphinx

### LINKED ISSUE #170: Migrate documenation to sphinx
## Summary
Migrate entire documentation to [sphinx](https://www.sphinx-doc.org/en/master/).

## Tasks
- [x] Add `sphinx` to dev dependencies
- [x] Add `furo` theme to dev dependencies
- [x] Add configuration 
- [x] Add nox target(s)
  - [x] build doc(s)
  - [x] open doc(s)
- [x] Add documentation to diliverable (wheel, tgz ...)
- [x] Add Github action
   - [x] Gh-pages branch with latest documentation
 -  [ ] ~~Create follow up ticket to host and publish multi version doc(s)~~

### PR DESCRIPTION
**Pull request by Nicoretti**
fixes #170 

- Add sphinx to dev dependencies
- Add sphinx-copybutton as dev dependency
- Add basic sphinx based documentation
- Add github action to generate documentation
- Include link to documentation in package

### CODE REVIEW

File: .github/workflows/gh-pages.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        run: |
+          poetry install
+      - name: Build Documenations
+        run: |
+          poetry run python -m nox -s build-docs
+          touch doc/build/html/.nojekyll
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@v4.3.0
+        with:
+          branch: gh-pages
```
Comment (tkilias) [inline .github/workflows/gh-pages.yaml:33]: Can this action do multiple versions? Why don't you use sphinx-github-pages generator

File: noxfile.py [hunk 2]
```diff
@@ -417,3 +420,32 @@ def create_parser() -> ArgumentParser:
...
+@nox.session(python=False, name="open-docs")
+def open_docs(session: Session) -> None:
+    """Open the documentation in the browser"""
+    index_page = DOC_BUILD / "index.html"
+    if not index_page.exists():
+        session.error(
+            f"File {index_page} does not exist." "Please run `nox -s build-docs` first"
+        )
+    session.run(
+        "python", "-m", "webbrowser", "-t", f"{index_page.resolve()}", external=True
```
Comment (tkilias) [inline noxfile.py]: You can use the browser module directly from python without subprocess

File: noxfile.py [hunk 3]
```diff
@@ -417,3 +420,32 @@ def create_parser() -> ArgumentParser:
...
+    """Remove all documentation artifacts"""
+    if DOC_BUILD.exists():
+        rmtree(DOC_BUILD.resolve())
+        session.log(f"Removed {DOC_BUILD}")
+
+
+@nox.session(python=False, name="build-docs")
+def build(session: Session) -> None:
+    """Build the documentation"""
+    session.run(
```
Comment (tkilias) [inline noxfile.py:437]: Do we also want to generate API doc or doesn't make this sense for this project?

**Review by tkilias — CHANGES_REQUESTED** [review]

File: .github/workflows/gh-pages.yaml (hunk 1 above)
Comment (Nicoretti) [reply .github/workflows/gh-pages.yaml:33]: wanted to have it as a to step thing, make sure the sphinx stuff is working, shown as expected then migrate.
I'll create a "follow" up ticket once this is merged.

Comment (Nicoretti) [reply .github/workflows/gh-pages.yaml:33]: @tkilias I don't thinks so, just took it from "my" standard sphinx github action workflow. As a first step I just wanted to introduce sphinx and the "happy" path I already know, then migrate to the `sphinx-github-pages generator`. Once this is merged I'll planed to create a follow up, see https://github.com/exasol/sqlalchemy-exasol/issues/170#issue-1316667931.

File: noxfile.py (hunk 3 above)
Comment (Nicoretti) [reply noxfile.py:437]: Currently we don't, was thinking of doing it but my thought was one would rather check the `sqla` documentation. The `sqlalchemy-exasol` code, is/should never be invoked directly by the `user`.

File: .github/workflows/gh-pages.yaml (hunk 1 above)
Comment (tkilias) [reply .github/workflows/gh-pages.yaml:33]: Ok, Sound good

**Review by tkilias — APPROVED** [review]

**Pull request merged**